### PR TITLE
Implement backpack.fetch console command

### PR DESCRIPTION
**Pull request by WheteThunger**
Allow players to move specific items from their backpack to their inventory
with a console command.

### CODE REVIEW

**Comment by LaserHydra** [comment]
Thank you for your contribution.

I am not opposed to adding the backpack.fetch command, however I feel like it maybe doesn't quite fit how the plugin is supposed to be used. Even then, it'd probably be an advanced feature not many people make use of, which would provide an advantage to certain players.
As a compromise, I am okay with this as long as it can be enabled/disabled in the config file. Maybe even default to disabled.

Since the backpack.fetch command does do some of the same checks as when opening a backpack, please create one "CanOpenBackpack" method which can then be used by both the command and when opening a backpack the usual way so we don't repeat ourselves.

Finally, please move any spacing/formatting changes to a separate PR (maybe together with the variable rename from https://github.com/LaserHydra/Backpacks/pull/7), or at least a separate commit.

**Comment by WheteThunger** [comment]
- I added a separate `backpacks.fetch` permission instead of a config option so it can be separately granted since it can provide an advantage for more savvy players
- I have undone the formatting changes that my IDE originally imposed

**Comment by WheteThunger** [comment]
Based on some older threads on the Backpacks support forum, I thought it would be wise to test if this can leak entities. I did some testing and found no leaks.
- Successfully transferring a full stack did not increase entity count when repeated many times.
- Splitting a stack and successfully transferring it did not entity count when repeated many times.
- Splitting a stack, unsuccessfully transferring that stack to the inventory, and forgetting about the split stack did not increase entity count when repeated many times. Seems it gets garbage collected  and the game properly cleans it up.

**Comment by WheteThunger** [comment]
Rebased and renamed `_CanOpenBackpack` -> `VerifyCanOpenBackpack`.

File: Backpacks.cs [hunk 1]
```diff
@@ -353,6 +355,102 @@ private void OpenBackpackConsoleCommand(ConsoleSystem.Arg arg)
...
+            if (itemDefinition != null)
+            {
+                itemID = itemDefinition.itemid;
+            }
+            else
+            {
+                // User may have provided an itemID instead of item short name
+                try
+                {
+                    itemID = Int32.Parse(itemArg);
```
Comment (LaserHydra) [inline Backpacks.cs]: Please use `int.TryParse` here.

File: Backpacks.cs [hunk 2]
```diff
@@ -353,6 +355,102 @@ private void OpenBackpackConsoleCommand(ConsoleSystem.Arg arg)
...
+                {
+                    PrintToChat(player, lang.GetMessage("Invalid Item", this, player.UserIDString));
+                    return;
+                }
+            }
+
+            int desiredAmount;
+            try
+            {
+                desiredAmount = Int32.Parse(args[1]);
```
Comment (LaserHydra) [inline Backpacks.cs]: Please use `int.TryParse` here too.

**Review by LaserHydra — CHANGES_REQUESTED** [review]
Good work, please just use `int.TryParse` where indicated and this should be good to go.

**Comment by LaserHydra** [comment]
Good work, thanks again!

**Pull request merged**